diff --git a/modules/simpletest/simpletest.test b/modules/simpletest/simpletest.test
index aa9a76e..e8241c2 100644
--- a/modules/simpletest/simpletest.test
+++ b/modules/simpletest/simpletest.test
@@ -44,7 +44,7 @@ class SimpleTestFunctionalTest extends DrupalWebTestCase {
     if (!$this->inCURL()) {
       $this->drupalGet('node');
       $this->assertTrue($this->drupalGetHeader('Date'), t('An HTTP header was received.'));
-      $this->assertTitle(variable_get('site_name', 'Drupal'), t('Site title matches.'));
+      $this->assertTitle(variable_get('site_name', 'Modified D6'), t('Site title matches.'));
       $this->assertNoTitle('Foo', t('Site title does not match.'));
       // Make sure that we are locked out of the installer when prefixing
       // using the user-agent header. This is an important security check.
